src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -457,6 +458,11 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
         holder.createErrorAnnotation(ret, "`return;` in a function whose return type is not `()` [E0069]")
     }
 
+    private fun checkExternCrate(holder: AnnotationHolder, el: RsExternCrateItem) {
+        if (el.reference.resolve() != null) return
```

Comment:
>  what do you mean by "setting origin = WORKSPACE" ?

I mean adding an additional check here:

```rust
if (el.containingCargoPackage?.origin != PackageOrigin.WORKSPACE) return
```

It will limit the scope of the check by our own project and prevent it from running within dependencies because we can't fully rely on correct resolving there as I mentioned earlier.